web/src/app/core/services/repository.service.ts

```diff
@@ -51,4 +51,27 @@ export class RepositoryService {
     const callable: any = this.fns.httpsCallable('deleteGitWebhookRepository');
     return callable({ repositoryUid, token: this.authService.profile.oauth.githubToken });
   }
+
+  public getRepoRating(repo: RepositoryModel): number {
+    let rating: number = 0;
+    const issuePoints: number = repo.issues.length > 0 ? this.getPoints(repo.issues[0].createdOn) : 0;
+    const releasesPoints: number = repo.releases.length > 0 ? this.getPoints(repo.releases[0].createdOn) : 0;
+    rating = (issuePoints + releasesPoints) * 5;
+    return rating;
+  }
+
+  public getPoints(dateStr: Date): number {
+    const date: Date = new Date();
+    const referenceDate: Date = new Date(dateStr);
+    let lapse: number = Math.floor((date.getTime() - referenceDate.getTime()) / 1000);
+    const day: number = 24 * 60 * 60;
+    const duration: number = lapse / day;
+    if (duration < 7) {
```

Comment:
A `switch` might be better here

Reply:
calculation logic changed so we don't need the switch here